nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -9,26 +9,134 @@
 from nncf.openvino.graph.node_utils import get_const_value
-from nncf.openvino.graph.node_utils import get_matmul_channel_axes
+from nncf.openvino.graph.node_utils import get_weight_channel_axes
 from nncf.parameters import CompressWeightsMode
+from nncf.quantization.algorithms.smooth_quant.backend import ALGO_BACKENDS
+from nncf.quantization.algorithms.weight_compression.backend import WeightCompressionAlgoBackend
 from nncf.quantization.fake_quantize import calculate_scale_zero_point
 
+
+@ALGO_BACKENDS.register(BackendType.OPENVINO)
+class OVWeightCompressionAlgoBackend(WeightCompressionAlgoBackend):
+    @property
+    def weighted_metatypes(self) -> List[OperatorMetatype]:
+        return [OVMatMulMetatype, OVEmbeddingMetatype]
+
+    @staticmethod
+    def is_node_with_weights(node: NNCFNode) -> bool:
+        return node.layer_attributes and node.layer_attributes.constant_attributes
+
+    @staticmethod
+    def validate_params(mode: CompressWeightsMode) -> None:
+        pass
+
+    @staticmethod
+    def do_compression(
+        model: ov.Model,
+        nodes_to_compress: List[NNCFNode],
+        mode: CompressWeightsMode,
+        ratio: float = None,
+        group_size: int = None,
+    ) -> ov.Model:
+        """
+        Compresses weights of Linear and Embedding layers to 8-bit integer or to nf4
+        depending on mode, ratio and group size.
+
+        :param model: The OpenVINO model for applying weight compression.
+        :param nodes_to_compress: List of nodes in the model's graph,
+            corresponding to the layers for weight compression.
+        :param mode: Defines a mode for weight compression.
+            INT8 stands for 8-bit integer quantization of all weights.
+            NF4 stands for a mixed-precision weights quantization to NF4 data type. The first and last layers
+            are always compressed to a backup precision which is 8-bit integer by default. All others are quantized
+            whether to NF4 or to a backup precision depending on criteria and the given ratio.
+        :param ratio: the ratio between baseline and backup precisions (e.g. 0.9 means 90% of layers quantized to NF4
+            and the rest to INT8).
+        :param group_size: number of weights (e.g. 128) in the channel dimension that share
+            quantization parameters (scale). The value -1 means no grouping.
+        """
+        all_weight_params: List[WeightNodeParams] = []
+        quantized_nodes_ids = set()
+
+        friendly_name_to_op_map = {op.get_friendly_name(): op for op in model.get_ops()}
+
+        for nncf_node in nodes_to_compress:
+            weight_port_ids = nncf_node.layer_attributes.get_const_port_ids()
+            for weight_port_id in weight_port_ids:
+                weight_op_friendly_name = nncf_node.layer_attributes.constant_attributes[weight_port_id]["name"]
+                weight_node = friendly_name_to_op_map[weight_op_friendly_name]
+                if weight_node is None:
```

Comment:
@l-bat please pay attention to the way of getting `weight_node` and manipulation with `port_id`
previously, it was different:
```python
allowed_metatypes_to_const_port = {OVEmbeddingMetatype: [0], OVMatMulMetatype: [0, 1]}

for node in model.get_ordered_ops():
    for const_port_id in allowed_metatypes_to_const_port[metatype]:
        weight_node = get_operation_const_op(node, const_port_id)
```